### PR TITLE
Migrated to axum 0.7

### PR DESCRIPTION
**Pull request by ItsEthra**
We just updated from rocket to axum and now we need to update to axum 0.7 🥲 oh well. ~~Right now mostly blocked by reqwest~~

### CODE REVIEW

**Comment by ItsEthra** [comment]
Workflow must be fixed to not run on draft prs I believe

**Comment by secana** [comment]
Wow, you are fast :)

The pipeline runs only the tests. Build/Deploy is skipped and only run if a "Github Release" is triggered. Thanks again, for putting in the work!

**Comment by ItsEthra** [comment]
Alright thanks, sorry for confusion I don't have much experience with github actions.

File: crates/auth/src/auth_req_token.rs [hunk 1]
```diff
@@ -220,10 +221,11 @@ mod auth_middleware_tests {
             }),
             ..appstate::test_state().await
         };
+
         Router::new()
-            .route("/guarded", get(handler_return_200))
+            .route("/guarded", get(StatusCode::OK))
```
Comment (jplatte) [inline crates/auth/src/auth_req_token.rs:222]: Nice to see the handler impl for `T: IntoResponse + Clone` get some use! Any reason you didn't replace the last remaining use of `handler_return_200` as well?

File: crates/web_ui/src/user.rs [hunk 2]
```diff
@@ -175,8 +178,8 @@ pub async fn logout(
         None => return Ok(jar), // Already logged out as no cookie can be found
     };
 
-    jar = jar.remove(Cookie::named(COOKIE_SESSION_ID));
-    jar = jar.remove(Cookie::build(COOKIE_SESSION_USER, "").path("/").finish());
+    jar = jar.remove(Cookie::build(COOKIE_SESSION_ID));
```
Comment (jplatte) [inline crates/web_ui/src/user.rs]:
```suggestion
    jar = jar.remove(COOKIE_SESSION_ID);
```

This should be the same, right?

File: crates/auth/src/auth_req_token.rs (hunk 1 above)
Comment (ItsEthra) [reply crates/auth/src/auth_req_token.rs:222]: I was just trying to fix the errors, I guess it's a good idea to replace other uses of it, will do.

**Pull request merged**